collector/lib/ResourceSelector.h

```diff
@@ -0,0 +1,35 @@
+//
+// Created by Robby Cochran on 3/28/24.
+//
+
+#ifndef RESOURCESELECTOR_H
+#define RESOURCESELECTOR_H
+#include <regex>
+#include <string>
+
+#include "storage/resource_collection.pb.h"
+
+#include "Hash.h"
+
+namespace collector {
+
+class ResourceSelector {
```

Comment:
The use of a class here feels a little redundant since everything is static anyway. A namespace might be a better fit